File: leetcode/35/35.cpp
#include <cstdio>
#include <vector>
using namespace std;

int searchInsert(vector<int>& nums, int target) {
        int size = nums.size();
        int left= 0, right = size-1;
        int middle =0;
        while(left <= right)
        {
            middle = (left+right)/2;

            if(target < nums[middle])
            {
                right = middle - 1;
            }
            else if(target > nums[middle])
            {
                left = middle + 1;
            }
            else return middle;
        }
        if(target < nums[middle])
        {
            return middle;
        }
        else return middle+1;
}

int main()
{
    vector<int> v = {1,3,5,6};
    int index = searchInsert(v,0);

    printf("%d\n", index);
}

File: leetcode/35/35.py
from typing import List, int
def searchInsert(nums: List[int], target: int)-> int:
    left = 0
    right = len(nums) -1
    middle = 0
    while left <= right :
        middle = (left + right) // 2
        if target < nums[middle] :
            right = middle - 1
        elif target > nums[middle] :
            left = middle + 1
        else :
            return nums[middle]
    if target > nums[middle] :
        return middle +1
    else:
        return middle